engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -375,6 +380,7 @@ public static boolean addAppender(String name, Properties properties)
       try {
         Method method = PropertyConfigurator.class.getDeclaredMethod("parseAppender", Properties.class, String.class);
         method.setAccessible(true);
+        makeMDCProperties();
```

Comment:
Ok. I've moved the initialization of the static appender properties to the static block. Also the method makeMDCProperties is public now. It will allow to call the initialization of the static appender properties explicitly (for instance, in unit tests). Also the new public method setServiceName(String service) is provided to set the service name explicitly.